test/system/projects/samples_test.rb

```diff
@@ -41,1291 +199,1535 @@ class SamplesTest < ApplicationSystemTestCase
       assert_text I18n.t(:'action_policy.policy.project.sample_listing?', name: @project.name)
     end
 
-    test 'should create sample' do
+    test 'create sample' do
+      ### SETUP START ###
       visit namespace_project_samples_url(@namespace, @project)
-      assert_selector 'a', text: I18n.t('projects.samples.index.new_button'), count: 1
+      within('#samples-table table tbody') do
+        # sample does not currently exist
+        assert_no_text 'New Name'
+      end
+      ### SETUP END ###
+
+      ### ACTIONS START ###
+      # launch dialog
       click_on I18n.t('projects.samples.index.new_button')
 
-      fill_in I18n.t('activerecord.attributes.sample.description'), with: @sample1.description
+      # fill new sample fields
+      fill_in I18n.t('activerecord.attributes.sample.description'), with: 'A sample description'
       fill_in I18n.t('activerecord.attributes.sample.name'), with: 'New Name'
       click_on I18n.t('projects.samples.new.submit_button')
+      ### ACTIONS END ###
 
+      ### VERIFY START ###
+      # success flash msg
       assert_text I18n.t('projects.samples.create.success')
-      assert_text 'New Name'
-      assert_text @sample1.description
+      # verify redirect to sample show page after successful sample creation
+      assert_selector 'h1', text: 'New Name'
+      assert_selector 'p', text: 'A sample description'
+      # verify sample exists in samples table
+      visit namespace_project_samples_url(@namespace, @project)
+      within('#samples-table table tbody') do
+        assert_text 'New Name'
```

Comment:
**nitpick (blocking):** This could be potentially flakey as there is no assertion to ensure that the table has been loaded on the page.

Reply:
Added in d60b9028eeca99d5aad54b37f1e7ce872ceb8dac